src/ring_theory/filtration.lean

```diff
@@ -7,13 +7,32 @@ Authors: Andrew Yang
 import ring_theory.noetherian
 import ring_theory.rees_algebra
 import ring_theory.finiteness
+import data.polynomial.module
+import order.hom.complete_lattice
 
 /-!
 
 # `I`-filtrations of modules
 
 This file contains the definitions and basic results around (stable) `I`-filtrations of modules.
 
+## Main results
+
+- `ideal.filtration`: An `I`-filtration on the module `M` is a sequence of decreasing submodules
+  `N i` such that `I • N ≤ I (i + 1)`. Note that we do not require the filtration to start from `⊤`.
+- `ideal.filtration.stable`: An `I`-filtration is stable if `I • (N i) = N (i + 1)` for large
```

Comment:
Should this be using the `filter.eventually` API? (Follow-up PR, I guess.)

Reply:
I think I tried this when defining `ideal.filtration.stable` but it caused some trouble without giving much benefit. I don't remember the detailed reasons though.